ctru-rs/src/gfx.rs

```diff
@@ -97,40 +119,90 @@ impl Gfx {
     pub fn wait_for_vblank(&self) {
         gspgpu::wait_for_event(gspgpu::Event::VBlank0, true);
     }
+}
+
+impl TopScreen {
+    /// Enable or disable the 3D stereoscopic effect
+    pub fn set_3d_enabled(&mut self, enabled: bool) {
+        unsafe {
+            ctru_sys::gfxSet3D(enabled);
+        }
+    }
 
-    /// Gets the framebuffer format for a screen
-    pub fn get_framebuffer_format(&self, screen: Screen) -> FramebufferFormat {
-        unsafe { ctru_sys::gfxGetScreenFormat(screen.into()).into() }
+    /// Enable or disable the wide screen mode (top screen).
+    /// This only works when 3D is disabled.
+    pub fn set_wide_mode(&mut self, enabled: bool) {
+        unsafe {
+            ctru_sys::gfxSetWide(enabled);
+        }
     }
 
-    /// Change the framebuffer format for a screen
-    pub fn set_framebuffer_format(&self, screen: Screen, fmt: FramebufferFormat) {
-        unsafe { ctru_sys::gfxSetScreenFormat(screen.into(), fmt.into()) }
+    /// Get the status of wide screen mode.
+    pub fn get_wide_mode(&self) -> bool {
+        unsafe { ctru_sys::gfxIsWide() }
     }
+}
 
-    /// Returns a tuple containing a pointer to the specifified framebuffer (as determined by the
-    /// provided `Screen` and `Side`), the width of the framebuffer in pixels, and the height of
-    /// the framebuffer in pixels
-    ///
-    /// Note that the pointer returned by this function can change after each call to this function
-    /// if double buffering is enabled
-    pub fn get_raw_framebuffer(&self, screen: Screen, side: Side) -> (*mut u8, u16, u16) {
+impl Screen for TopScreen {
+    fn into_raw(&self) -> ctru_sys::gfxScreen_t {
+        ctru_sys::GFX_TOP
+    }
+
+    fn set_double_buffering(&mut self, enabled: bool) {
+        unsafe { ctru_sys::gfxSetDoubleBuffering(ctru_sys::GFX_TOP, enabled) }
+    }
```

Comment:
I wonder if some of these could have default trait impls since they just make a call using the result of `self.into_raw()`. Would just

Reply:
This is a good idea. We can use `into_raw` to get the `gfxScreen_t` value.